### DISCUSSION TITLE
Materialized View not following given limit.

### DISCUSSION THREAD

**Discussion by KRISHNA-PRAVESH** [ClickHouse/ClickHouse#70632] [Q&A] [answered]
Hi, I am new clickhouse and currently exploring it. 
I have a table that stores events. I have created a materialized view for that table that has the latest logs (let's say latest 10 logs) like the following,

`create materialized view latest_events_mv engine = MergeTree() order by creation_time populate as select * from events order by time desc limit 10;`

This materialized view works well the first time I create it, but after I insert a new record to the main table , the same record is updated in the materialized view also but now the materialized view is having 11 rows. 

Could any one explain why the materialized view is not following the limit I have given during its creation and is there any way to maintain the limit to store the latest logs. 

Thanks in Advance.

**Comment by den-crane** [comment, marked as answer]
That is expected because of the nature of Materialized View https://den-crane.github.io/Everything_you_should_know_about_materialized_views_commented.pdf

Try [Refreshable Materialized View ](https://clickhouse.com/docs/en/sql-reference/statements/create/view#refreshable-materialized-view)

**Reply by KRISHNA-PRAVESH** [reply]
Thanks for the response. I'll check that out.
